console/network/environment/src/helpers/sanitizer.rs

```diff
@@ -117,12 +117,33 @@ impl Sanitizer {
     ///
     /// Discard any leading newline.
     fn str_till_eol(string: &str) -> ParserResult<&str> {
-        map(
-            recognize(Self::till(alt((value((), tag("\\\n")), value((), Sanitizer::parse_safe_char))), Self::eol)),
-            |i| {
-                if i.as_bytes().last() == Some(&b'\n') { &i[0..i.len() - 1] } else { i }
-            },
-        )(string)
+        // A heuristic approach is applied here in order to avoid
+        // costly parsing operations in the most common scenarios.
```

Comment:
Nit. Can you explain the heuristic in the comment?

Reply:
added to my TODO (as with the `Program`-related comment, I'd suggest to add it separately to avoid merge-related issues)